GovUkDesignSystem/Helpers/HtmlGenerationHelpers.cs

```diff
@@ -76,20 +76,27 @@ public static void SetErrorMessages(IHasErrorMessage target, ModelStateEntry mod
         public static List<TEnum> GetListOfEnumValuesFromModelStateOrModel<TModel, TEnum>(
             TModel model,
             Expression<Func<TModel, List<TEnum>>> propertyLambdaExpression,
-            ModelStateEntry modelStateEntry)
+            ModelStateEntry modelStateEntry,
+            string ignoreValue = null)
             where TModel : class
             where TEnum : struct, Enum
         {
             if (modelStateEntry != null && modelStateEntry.RawValue != null)
             {
+                var stringValues = new List<string>();
                 if (modelStateEntry.RawValue is string[])
                 {
-                    return ((string[])modelStateEntry.RawValue).Select(e => (TEnum)Enum.Parse(typeof(TEnum), e.ToString())).ToList();
+                    stringValues.AddRange((string[])modelStateEntry.RawValue);
                 }
                 else if (modelStateEntry.RawValue is string)
                 {
-                    return new List<TEnum> { (TEnum)Enum.Parse(typeof(TEnum), modelStateEntry.RawValue.ToString()) };
+                    stringValues.Add((string)modelStateEntry.RawValue);
                 }
+
+                return stringValues
+                    .Except(new[] { ignoreValue })
+                    .Select(e => (TEnum)Enum.Parse(typeof(TEnum), e.ToString()))
```

Comment:
This is the bit that was actually broken as the dummy value would fail to parse.